qa: test_rbd_wnbd.py - disable PS progress bar

petrutlucian94 opened the pull request:
This test uses certain PS commands that attempt to display a progress bar. However, this can cause issues when invoked remotely (e.g. by the jenkins job).

For this reason, we're defining a helper (ps_execute) that runs PS commands, disabling the progress bars and enabling the non interactive mode.

Signed-off-by: Lucian Petrut <lpetrut@cloudbasesolutions.com>


<!--
  - Please give your pull request a title like

      [component]: [short description]

  - Please use this format for each git commit message:

      [component]: [short description]

      [A longer multiline description]

      Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
      Signed-off-by: [Your Name] <[your email]>

    For examples, use "git log".
-->

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [x] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

idryomov commented on qa/workunits/windows/test_rbd_wnbd.py:
```diff
@@ -376,36 +382,32 @@ def cleanup(self):
-             f"(Get-Partition -DiskNumber {self.disk_number}"
-             " | ? { $_.DriveLetter }) | Format-Volume -Force -Confirm:$false")
-        execute(*cmd)
+            f"(Get-Partition -DiskNumber {self.disk_number}"
```
Nit: This could go after `cmd = (`, on the same line.

petrutlucian94 replied:
The second line would exceed 80 characters:

```
def _format_volume(self):
        cmd = (f"(Get-Partition -DiskNumber {self.disk_number}"
               " | ? { $_.DriveLetter }) | Format-Volume -Force -Confirm:$false")
```

The pep8 linter wouldn't be happy with the following indentation:

```
def _format_volume(self):
        cmd = (f"(Get-Partition -DiskNumber {self.disk_number}"
            " | ? { $_.DriveLetter }) | Format-Volume -Force -Confirm:$false")
```

We'd have to either keep this or add a third line:

```
def _format_volume(self):
        cmd = (f"(Get-Partition -DiskNumber {self.disk_number}"
               " | ? { $_.DriveLetter }) | "
               "Format-Volume -Force -Confirm:$false")
```

That's why I usually prefer starting long multi-line strings from a new line, so that I get a little more headroom for the string by avoiding some extra indents.

idryomov replied:
> The second line would exceed 80 characters:

My apologies, I commented on the wrong line.  This comment was supposed to go for the one in `_get_drive_letter`.  This one, in `_format_volume`, was better with your original formatting.

Do you want to revert this hunk and amend the one in `_get_drive_letter` or just punt on this?

idryomov replied:
`make check` needs to be rerun anyway -- might as well be after another force push ;)

idryomov commented:
The check is failing due to on-going lab connectivity/VPN issues:
```
+ mkdir -p /home/jenkins-build/build/workspace/ceph-windows-pull-requests/libvirt
+ echo 'Downloading VM image from https://filedump.ceph.com/windows/ceph-win-ltsc2019-ci-image.qcow2'
Downloading VM image from https://filedump.ceph.com/windows/ceph-win-ltsc2019-ci-image.qcow2
+ curl -s -L https://filedump.ceph.com/windows/ceph-win-ltsc2019-ci-image.qcow2 -o /home/jenkins-build/build/workspace/ceph-windows-pull-requests/libvirt/ceph-windows-client.qcow2
Build step 'Execute shell' marked build as failure
```

idryomov commented:
jenkins test windows

petrutlucian94 commented:
I see that https://filedump.ceph.com/ is still down. IMO, we could go ahead and merge it, it might take a while until the file server comes back up.

I've tested it locally, plus there were just a few minor formatting changes since the last successful CI run.

idryomov commented:
I wanted to see the check pass again but, yeah, it might take a while.  Merging this so that https://github.com/ceph/ceph/pull/48848 can be properly rebased -- I saw that this commit got pulled there.
